RestApi/Sepes.RestApi/Services/PodService.cs

```diff
@@ -115,46 +115,37 @@ private async Task ManageNetworkSecurityGroup(Pod newPod, Pod based)
 
             // pod.allowAll != basePod.allowAll // Apply allow all
 
             // Set inbound and outbound rules
-            var addRuleTasks = new List<Task>();
-            if (based == null || !based.incoming.SequenceEqual(newPod.incoming))
-            {
-                Dictionary<ushort, string[]> inbound = GenerateRuleDictionary(newPod.incoming);
-                int priority = 100;
-
-                foreach (var port in inbound.Keys) {
-                    await _azure.NsgAllowInboundPort(nsgName, newPod.resourceGroupName, "Port_" + port, priority++, inbound[port], (int) port);
-                    Console.WriteLine("#### Added rule inbound "+port+" "+inbound[port].ElementAt(0));
-                }
+            Dictionary<ushort, string[]> inbound = GenerateRuleDictionary(newPod.incoming);
+            int inPriority = 100;
+            foreach (var port in inbound.Keys) {
+                await _azure.NsgAllowInboundPort(nsgName, newPod.resourceGroupName, "Port_" + port, inPriority++, inbound[port], (int) port);
+                Console.WriteLine("#### Added rule inbound "+port+" "+inbound[port].ElementAt(0));
             }
-            if (based == null || !based.outgoing.SequenceEqual(newPod.outgoing))
-            {
-                Dictionary<ushort, string[]> outbound = GenerateRuleDictionary(newPod.outgoing);
-                int priority = 100;
-
-                foreach (var port in outbound.Keys) {
-                    await _azure.NsgAllowOutboundPort(nsgName, newPod.resourceGroupName, "Port_" + port, priority++, outbound[port], (int) port);
-                    Console.WriteLine("#### Added rule outbound "+port+" "+outbound[port].ToString());
-                }
+        
+            Dictionary<ushort, string[]> outbound = GenerateRuleDictionary(newPod.outgoing);
+            int outPriority = 100;
+            foreach (var port in outbound.Keys) {
+                await _azure.NsgAllowOutboundPort(nsgName, newPod.resourceGroupName, "Port_" + port, outPriority++, outbound[port], (int) port);
+                Console.WriteLine("#### Added rule outbound "+port+" "+outbound[port].ElementAt(0));
             }
 
+
             // Apply network security group to subnet
             Console.WriteLine($"#### ApplySecurityGroup() {newPod.resourceGroupName}, {nsgName}, {newPod.subnetName}, {newPod.networkName}");
             await _azure.ApplySecurityGroup(newPod.resourceGroupName, nsgName, newPod.subnetName, newPod.networkName);
 
             // Delete old nsg
             if (nsgName == nsgNameDefault && nsgNames.Contains(nsgNameDef2)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDef2);
                 await _azure.DeleteSecurityGroup(nsgNameDef2, newPod.resourceGroupName);
             }
             else if (nsgName == nsgNameDef2 && nsgNames.Contains(nsgNameDefault)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDefault);
```

Comment:
Console.WriteLine should not be in master. If debug remove. If not log it useing logger.